### PR TITLE
feat: change offerArgs keyword from want to maxBuy, wire through CLI

### LINKED ISSUE #7329: Auction should use `maxBuy` instead of `want` in offerArgs
## What is the Problem Being Solved?

The auction's offerArgs require specifying `want` to indicate how much collateral a bid desires to purchase. This is a conflict with the `want` in Zoe's proposals.

## Description of the Design

Rename it to `maxBuy`.

Recently updated via conversation here https://agoricopco.slack.com/archives/C038TM0EUKG/p1681401330392579

## Security Considerations

reduce confusion

## Scaling Considerations

None

## Test Plan

update existing tests.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #7329

## Description

Change the offerArgs keyword used by the auction from `want` to `maxBuy`.  Plumb the changes through the CLI.

### Security Considerations

Reduced confusion.

### Scaling Considerations

None

### Documentation Considerations

The CLI has internal docs, which have been updated. Auction docs don't exist yet.

### Testing Considerations

Tests updated.

### CODE REVIEW

File: packages/agoric-cli/test/test-inter-cli.js [hunk 1]
```diff
@@ -222,11 +222,11 @@ const offerStatus2 = harden({
       denominator: { brand: topBrands.IST, value: 100n },
       numerator: { brand: topBrands.IST, value: 90n },
     },
-    want: { brand: topBrands.ATOM, value: 2000000n },
+    maxBuy: { brand: topBrands.ATOM, value: 2000000n },
```
Comment (dckc) [inline packages/agoric-cli/test/test-inter-cli.js:225]: :clap: nice catch!

**Review by dckc — APPROVED** [review]
looks right on target

File: packages/agoric-cli/test/test-inter-cli.js [hunk 2]
```diff
@@ -222,11 +222,11 @@ const offerStatus2 = harden({
...
   },
   proposal: {
     give: {
-      [KW.Bid]: { brand: topBrands.ATOM, value: 20000000n },
+      [KW.Bid]: { brand: topBrands.IST, value: 20000000n },
```
Comment (dckc) [inline packages/agoric-cli/test/test-inter-cli.js:229]: er... this is the nice catch I meant

File: packages/inter-protocol/test/test-clientSupport.js [hunk 3]
```diff
@@ -101,7 +101,7 @@ test('Offers.auction.Bid', async t => {
   t.deepEqual(
     Offers.auction.Bid(brands, {
       offerId: 'by-price2',
-      desiredBuy: '10_000ATOM',
+      maxBuy: '10_000ATOM',
```
Comment (dckc) [inline packages/inter-protocol/test/test-clientSupport.js:104]: looks like there's one more `want`->`maxBuy` to fix on line 123

File: packages/agoric-cli/test/test-inter-cli.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/agoric-cli/test/test-inter-cli.js:229]: That's what I thought you must have been talking about. 😄

File: packages/inter-protocol/test/test-clientSupport.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/test/test-clientSupport.js:104]: done. Thanks.

**Pull request merged**